src/v/datalake/datalake_manager.cc

```diff
@@ -178,6 +187,10 @@ ss::future<> datalake_manager::start() {
       = _partition_mgr->local().register_unmanage_notification(
         model::kafka_namespace, [this](model::topic_partition_view tp) {
             model::ntp ntp{model::kafka_namespace, tp.topic, tp.partition};
+            // We remove the probe only when partition is moved out of the
+            // shard to avoid metrics disappearing during much more common
+            // leadership changes.
+            _translation_probe_by_ntp.erase(ntp);
```

Comment:
Hm I actually thought having the partition probe being the parent was pretty elegant. Curious if you ran up against something that made it impractical? If the concern was about generalizability, I'm sure we could've just put the translation probe into the cluster namespace or something.

Managing the probe in the datalake manager is fine too, but it does feel like there's a bit more overhead in making sure it's correct.

Reply:
@andrwng couple of reasons for switching to the current approach:
1) reviewing partition probe (part) was a bit more complex than this and i didn't want to introduce that complexity now; i want to move a little bit faster
2) supporting dynamic label set began to put strain on the api design i began with; mainly because of some peculiarities in the partition probe itself (e.g. clear metrics api, reconfiguration on config change for schema registry) that meant that partition_probe_part needed more API surface or it could manage the metrics group itself but then almost all the benefits of putting it in the partition_probe were lost

I find this design much simpler.